interface/globals.php

```diff
@@ -130,8 +130,13 @@
         }
     }
 
+    // for both REST API and browser access we can't proceed unless we have a valid site id.
+    // since this is user provided content we need to escape the value but we use htmlspecialchars instead
+    // of text() as our helper functions are loaded in later on in this file.
     if (empty($tmp) || preg_match('/[^A-Za-z0-9\\-.]/', $tmp)) {
-        die("Site ID '". text($tmp) . "' contains invalid characters.");
+        echo "Invalid URL";
+        error_log("Request with site id '". htmlspecialchars($tmp, ENT_NOQUOTES) . "' contains invalid characters.");
```

Comment:
Don't need to escape stuff going to log:
```
error_log("Request with site id '" . $tmp . "' contains invalid characters.");
```
(that definitely will fix the text() issue :) )

Reply:
Is that the project preference? I've always in the past escaped stuff going to logs as there are php security exploits that can occur through php attackers remotely including the error_log.  Had to fix a compromised wordpress site where that occurred once.

Reply:
Here's OWASP's take on it: https://www.owasp.org/index.php/Log_Injection

Reply:
hi @adunsulag ,
I wasn't aware of that. The attacker is running the error log output through html?

Reply:
Interesting. Just added this to our Code Security page:
https://www.open-emr.org/wiki/index.php/Codebase_Security#Log_Injection
(html escaping variables that go to log is a good plan for now)

Reply:
thanks @adunsulag ! Code is ready for codebase.